packages/magic/src/index.ts

```diff
@@ -0,0 +1,198 @@
+          })
+        }
+
+        let magicProvider = magicInstance.rpcProvider
+        let provider: EIP1193Provider
+
+        function patchProvider(): EIP1193Provider {
+          provider = createEIP1193Provider(magicProvider, {
+            eth_requestAccounts: async ({ baseRequest }) => {
+              try {
+                if (!loggedIn) await handleLogin()
+                const accounts = await baseRequest({ method: 'eth_accounts' })
+                return accounts
+              } catch (error) {
+                console.error('error in request accounts', error)
+                const { code } = error as { code: number }
+                if (code === -32603) {
+                  throw new ProviderRpcError({
+                    code: ProviderRpcErrorCode.ACCOUNT_ACCESS_REJECTED,
+                    message: 'Account access rejected'
+                  })
+                }
+                return []
+              }
+            },
+            eth_selectAccounts: null,
+            eth_getBalance: async () => {
+              const [address] = (
+                await magicProvider.request({
+                  method: 'eth_accounts'
+                })
+              )
+              const balance = await magicProvider.request({
+                method: 'eth_getBalance',
+                params: [address, 'latest']
+              })
+              return balance
+                ? BigNumber.from(balance).mul('1000000000000000000').toString()
+                : '0'
+            },
+            wallet_switchEthereumChain: async ({ params }) => {
+              const chain = chains.find(({ id }) => id === params[0].chainId)
+              if (!chain) throw new Error('Chain must be set before switching')
+              currentChain = chain.id
+              // re-instantiate instance with new network
+              customNodeOptions = {
+                chainId: parseInt(chains[0].id, 10),
+                rpcUrl: chains[0].rpcUrl
+              }
+
+              magicInstance = new Magic(apiKey, {
+                network: customNodeOptions
+              })
+
+              magicProvider = magicInstance.rpcProvider
+
+              emitter.emit('chainChanged', chain.id)
+
+              patchProvider()
+
+              return null
+            },
+            eth_accounts: async ({ baseRequest }) => {
+              const accounts = await baseRequest({ method: 'eth_accounts' })
+              return Array.isArray(accounts) && accounts.length
+                ? [accounts[0]]
+                : []
+            },
+            eth_sign: async ({ params: [address, message] }) => {
+              const receipt = await magicProvider.send('eth_sign', [
+                address,
+                message
+              ])
+              return receipt &&
+                receipt.hasOwnProperty('tx') &&
+                receipt.tx.hasOwnProperty('hash')
+                ? receipt.tx.hash
+                : ''
+            },
```

Comment:
Might be able to shorten this method!?

```suggestion
            eth_sign: async ({ params }) =>  
              magicProvider.send('eth_sign', params).then( receipt => 
                receipt && receipt.hasOwnProperty('tx') && 
                receipt.tx.hasOwnProperty('hash') ? receipt.tx.hash : ''
              )
```

Reply:
I removed the destructuring since the params are in the the proper order for the calls but kept the await instead of the .then
Good call on the destructuring, I was thinking the same thing but wasn't sure if that would hurt readability and since you are one of the readers you confirmed 💯